### DISCUSSION TITLE
why windows pc screen was not locked when a rustdesk client connect to it by RDP?

### DISCUSSION THREAD

**Discussion by louislau86** [rustdesk/rustdesk#7916] [Q&A] [answered]
RustDesk, as a remote desktop software, allows users to access and control another computer remotely. Unlike mstsc, which is a native Windows feature that locks the remote computer's screen by default when a remote session is established, RustDesk does not enforce such a behavior. I wanna figure out the difference between the two ways, and how rustdesk implment this.

**Comment by dimejo** [comment]
RustDesk is an alternative for TeamViewer or Anydesk, which AFAIR do behave the same. When using this kind of software for technical support it sometimes is necessary for the user in front of the PC to see everything that is happening.

**Reply by louislau86** [reply]
Thanks for your reply! But I am still want to know how rustdesk implment the function that don't lock the screen of the remote windows PC? It's a difficult thing for me to read rust code.

**Reply by dimejo** [reply]
Sorry, can't help with code.

**Reply by louislau86** [reply]
Maybe rustdesk don't use rdp of windows? but i found the code call "mstsc" in function "run_rdp"

**Comment by louislau86** [comment]
@21pages  Sorry to disturb you. But i wanna figure out how rustdesk connect a remote windows by RDP and the controller windows pc won't lock the screen. Is there any setting or something else set by rustdesk ? Wish your response, thank you!

**Reply by 21pages** [reply]
RustDesk will not lock the screen when connected, you can use privacy mode to achieve similar functionality

**Reply by louislau86** [reply]
You mean use privacy mode when connect to a remote windows by "mstsc"?

**Reply by louislau86** [reply]
The desired effect is that when using RDP to connect to a remote Windows computer, the screen of the remote computer is not locked. But when I use MSTSC, the remote computer will default to locking the screen. But Rustdesk also applies the RDP protocol, but the remote computer does not lock the screen. I want to know how this is implemented. Thank you!

**Reply by 21pages** [reply, marked as answer]
RustDesk uses video codec, not ms rdp protocol.

**Reply by louislau86** [reply]
Rustdesk also support RDP when we connect a remote PC by direct connections.

**Reply by 21pages** [reply]
ms rdp is a session, usually windows only allow one session, so the physical console session is logged out

**Reply by louislau86** [reply]
Thank you, maybe i misunderstood the rustdesk behaviors.
